src/doc/src/SUMMARY.md

```diff
@@ -15,33 +15,33 @@
     * [Cargo.toml vs Cargo.lock](guide/cargo-toml-vs-cargo-lock.md)
     * [Tests](guide/tests.md)
     * [Continuous Integration](guide/continuous-integration.md)
+    * [Publishing on crates.io](guide/publishing.md)
```

Comment:
Redirections are needed. Maybe use this? https://rust-lang.github.io/mdBook/format/configuration/renderers.html#outputhtmlredirect

Reply:
> This will generate an HTML page which will automatically redirect to the given location. Note that the source location does not support # anchor redirects.

This is the interesting part of this config. We might need to do this as well?

https://github.com/rust-lang/cargo/blob/2fd7321ee49307c328e0414c85fb4c36a5653759/src/doc/src/reference/specifying-dependencies.md#L631-L648

Reply:
So instead of using the redirect feature, you are proposing that we place a hand-written file in the old location that includes the script to redirect fragments?  I'm not fully confident in making sure all of these features interact well with each other, search engines, etc.

Reply:
The mdbook config one is for generating those old pages with a `<meta http-equiv="refresh" content="0; URL=/.path/to/new/location.html">`, which is a reasonable way to getting a better search engine optimization. However it doesn't support hash. The hand-written `<script>` hack doesn't seem to help this situation, either.

Off the top of my head, the only way to support hash redirections is using JavaScript in the old location of the markdown file, and make the markdown file hidden. It is not good, though.

Reply:
I have another "bad idea": don't rename file names. Just swap the order in SUMMARY.md.

Reply:
cc @ehuss as you may know how well we need to maintain link validity. Like would tools in rust-lang/rust check this?

Reply:
Hm, yea, this might actually be a little bit of a problem. I think there are quite a few links to https://doc.rust-lang.org/cargo/reference/publishing.html#github-permissions in particular. 

I actually don't think there is a way to include a redirect file without leaving an entry in SUMMARY which I think is undesirable here.

I might recommend just moving the entries in SUMMARY.md, but don't rename the file. I realize that is not optimal, since the URL is a little confusing that way.

Perhaps in the future we can extend mdbook to support javascript redirects with HTML fragments on removed pages? We can rename the files at that time.

Reply:
Oh, I see what you mean by leaving an html file in its place. I *think* that should work. That seems like a reasonable option, too.

Reply:
For now, I'm leaving the path in the old location.